--- switch_rtp.c	2023-08-23 17:17:24.229732433 +0000
+++ switch_rtp.c.new	2023-08-23 17:41:05.504067569 +0000
@@ -5482,7 +5482,7 @@
 static switch_size_t do_flush(switch_rtp_t *rtp_session, int force, switch_size_t bytes_in)
 {
 	int was_blocking = 0;
-	switch_size_t bytes;
+	//switch_size_t bytes;
 	switch_size_t bytes_out = 0;

 	if (!switch_rtp_ready(rtp_session)) {
@@ -5542,7 +5542,7 @@
 #endif
 		    handle_rfc2833(rtp_session, bytes_in, &do_cng);
 		}
-
+/* DH: disable this flushing feature, would prefer to receive all packets
 		do {
 			if (switch_rtp_ready(rtp_session)) {
 				bytes = sizeof(rtp_msg_t);
@@ -5555,7 +5555,7 @@
 						rtp_session->last_media = switch_micro_time_now();
 					}

-					/* Make sure to handle RFC2833 packets, even if we're flushing the packets */
+					// Make sure to handle RFC2833 packets, even if we're flushing the packets
 					if (bytes > rtp_header_len && rtp_session->recv_msg.header.version == 2 && rtp_session->recv_msg.header.pt == rtp_session->recv_te) {
 						rtp_session->last_rtp_hdr = rtp_session->recv_msg.header;
 						handle_rfc2833(rtp_session, bytes, &do_cng);
@@ -5572,7 +5572,7 @@
 				break;
 			}
 		} while (bytes > 0);
-
+*/
 #ifdef DEBUG_2833
         if (flushed) {
             switch_log_printf(SWITCH_CHANNEL_LOG, SWITCH_LOG_ERROR, "*** do_flush: total flushed packets: %ld ***\n",(long)flushed);
