--- mod_avmd.c	2022-02-10 11:19:05
+++ mod_avmd.c.new	2023-04-19 13:28:03
@@ -1476,15 +1476,20 @@
     }
     if ((SWITCH_CALL_DIRECTION_OUTBOUND == switch_channel_direction(channel)) && (avmd_session->settings.outbound_channnel == 1)) {
             flags |= SMBF_READ_REPLACE;
-			direction = "READ_REPLACE";
+            direction = "READ_REPLACE";
     }
-    if ((SWITCH_CALL_DIRECTION_INBOUND == switch_channel_direction(channel)) && (avmd_session->settings.inbound_channnel == 1)) {
+    if ((SWITCH_CALL_DIRECTION_INBOUND == switch_channel_direction(channel)) /* && (avmd_session->settings.inbound_channnel == 1) */) {
+            /* DCH: for drachtio-fsmrf */
+            flags |= SMBF_READ_REPLACE;
+            direction = "READ_REPLACE";
+/*
             flags |= SMBF_WRITE_REPLACE;
 			if (!strcmp(direction, "READ_REPLACE")) {
 				direction = "READ_REPLACE | WRITE_REPLACE";
 			} else {
 				direction = "WRITE_REPLACE";
 			}
+*/
     }
 
     if (flags == 0) {
