--- switch_rtp.c	2021-01-12 02:11:42.334172596 +0000
+++ switch_rtp.c.new	2021-01-12 02:12:42.695207260 +0000
@@ -5639,8 +5639,8 @@
 static switch_size_t do_flush(switch_rtp_t *rtp_session, int force, switch_size_t bytes_in)
 {
 	int was_blocking = 0;
-	switch_size_t bytes;
-	uint32_t flushed = 0;
+	//switch_size_t bytes;
+	//uint32_t flushed = 0;
 	switch_size_t bytes_out = 0;
 
 	if (!switch_rtp_ready(rtp_session)) {
@@ -5700,7 +5700,7 @@
 #endif
 		    handle_rfc2833(rtp_session, bytes_in, &do_cng);
 		}
-
+/*
 		do {
 			if (switch_rtp_ready(rtp_session)) {
 				bytes = sizeof(rtp_msg_t);
@@ -5713,7 +5713,7 @@
 						rtp_session->last_media = switch_micro_time_now();
 					}
 
-					/* Make sure to handle RFC2833 packets, even if we're flushing the packets */
+					//Make sure to handle RFC2833 packets, even if we're flushing the packets 
 					if (bytes > rtp_header_len && rtp_session->recv_msg.header.version == 2 && rtp_session->recv_msg.header.pt == rtp_session->recv_te) {
 						rtp_session->last_rtp_hdr = rtp_session->recv_msg.header;
 						handle_rfc2833(rtp_session, bytes, &do_cng);
@@ -5732,7 +5732,7 @@
 				break;
 			}
 		} while (bytes > 0);
-
+*/
 #ifdef DEBUG_2833
         if (flushed) {
             switch_log_printf(SWITCH_CHANNEL_LOG, SWITCH_LOG_ERROR, "*** do_flush: total flushed packets: %ld ***\n",(long)flushed);
